WordPress/Classes/Models/Blog.m

```diff
@@ -439,6 +444,46 @@ - (BOOL)supports:(BlogFeature)feature
     }
 }
 
+- (BOOL)supportsPublicize
+{
+    // wpcom blog and publicize has not been disabled
+    if (self.isHostedAtWPcom && ![[self getOptionValue:OptionsKeyPublicizeDisabled] boolValue]) {
+        return YES;
+    }
+
+    // self-hosted connected to jetpack via the wpcom acct signed into the app.
+    if (self.dotComID && [self jetpackPublicizeModuleEnabled]) {
+        return YES;
+    }
```

Comment:
I see, I forgot the first check was `HostedAtWPcom` and not just REST. I still find the logic a bit hard to follow. How about this?

```obj-c
    // Publicize is only supported via REST
    if (![self supports:BlogFeatureWPComRESTAPI]) {
        return NO;
    }

    if (self.isHostedAtWPcom) {
        // For WordPress.com YES unless it's disabled
        return ![[self getOptionValue:OptionsKeyPublicizeDisabled] boolValue];
    } else {
        // For Jetpack, check if the module is enabled
        return [self jetpackPublicizeModuleEnabled];
    }
```